### PR TITLE
Add missing indices for `status` columns

### PR DESCRIPTION
**Pull request by perangel**
## What
Adds column indices to the following tables:
* `attempts.status`
* `connection.status`
* `jobs.status`

The offending query that was used to diagnose these was the following:

```
SELECT
  "public"."attempts"."processing_task_queue",
  COUNT(*) AS "count"
FROM
  "public"."jobs"
JOIN
  "public"."connection"
ON
  CAST("public"."connection"."id" AS varchar(255)) = "public"."jobs"."scope"
JOIN
  "public"."attempts"
ON
  "public"."attempts"."job_id" = "public"."jobs"."id"
WHERE
  ("public"."jobs"."status" = 'succeeded'::"public"."job_status"
    AND "public"."connection"."status" = 'active'::"public"."status_type"
    AND "public"."attempts"."status" = 'succeeded'::"public"."attempt_status")
GROUP BY
  "public"."attempts"."processing_task_queue"
;^C
prod-configapi=> explain analyze SELECT
  "public"."attempts"."processing_task_queue",
  COUNT(*) AS "count"
FROM
  "public"."jobs"
JOIN
  "public"."connection"
ON
  CAST("public"."connection"."id" AS varchar(255)) = "public"."jobs"."scope"
JOIN
  "public"."attempts"
ON
  "public"."attempts"."job_id" = "public"."jobs"."id"
WHERE
  ("public"."jobs"."status" = 'succeeded'::"public"."job_status"
    AND "public"."connection"."status" = 'active'::"public"."status_type"
    AND "public"."attempts"."status" = 'succeeded'::"public"."attempt_status")
GROUP BY
  "public"."attempts"."processing_task_queue"
;
                                                                               QUERY PLAN

---------------------------------------------------------------------------------------------------------------------------------------------------------------
---------
 Finalize GroupAggregate  (cost=251801.92..251802.68 rows=3 width=9) (actual time=1120.110..1142.062 rows=4 loops=1)
   Group Key: attempts.processing_task_queue
   ->  Gather Merge  (cost=251801.92..251802.62 rows=6 width=9) (actual time=1120.098..1142.047 rows=12 loops=1)
         Workers Planned: 2
         Workers Launched: 2
         ->  Sort  (cost=250801.90..250801.91 rows=3 width=9) (actual time=1114.887..1114.959 rows=4 loops=3)
               Sort Key: attempts.processing_task_queue
               Sort Method: quicksort  Memory: 25kB
               Worker 0:  Sort Method: quicksort  Memory: 25kB
               Worker 1:  Sort Method: quicksort  Memory: 25kB
               ->  Partial HashAggregate  (cost=250801.84..250801.87 rows=3 width=9) (actual time=1114.840..1114.911 rows=4 loops=3)
                     Group Key: attempts.processing_task_queue
                     Batches: 1  Memory Usage: 24kB
                     Worker 0:  Batches: 1  Memory Usage: 24kB
                     Worker 1:  Batches: 1  Memory Usage: 24kB
                     ->  Parallel Hash Join  (cost=128458.56..250083.22 rows=143724 width=1) (actual time=993.788..1093.514 rows=101793 loops=3)
                           Hash Cond: (attempts.job_id = jobs.id)
                           ->  Parallel Seq Scan on attempts  (cost=0.00..116235.59 rows=311791 width=9) (actual time=0.015..177.859 rows=249386 loops=3)
                                 Filter: (status = 'succeeded'::attempt_status)  <--------------- BOOOOO
                                 Rows Removed by Filter: 52203
                           ->  Parallel Hash  (cost=126045.85..126045.85 rows=147017 width=8) (actual time=731.137..731.140 rows=101793 loops=3)
                                 Buckets: 131072  Batches: 8  Memory Usage: 2560kB
                                 ->  Merge Join  (cost=119016.39..126045.85 rows=147017 width=8) (actual time=532.692..702.089 rows=101793 loops=3)
                                       Merge Cond: ((jobs.scope)::text = (((connection.id)::character varying(255))::text))
                                       ->  Sort  (cost=117666.88..118411.19 rows=297726 width=45) (actual time=522.635..618.178 rows=238282 loops=3)
                                             Sort Key: jobs.scope
                                             Sort Method: external merge  Disk: 11104kB
                                             Worker 0:  Sort Method: external merge  Disk: 11104kB
                                             Worker 1:  Sort Method: external merge  Disk: 18464kB
                                             ->  Parallel Seq Scan on jobs  (cost=0.00..81438.69 rows=297726 width=45) (actual time=0.013..106.547 rows=238282
loops=3)
                                                   Filter: (status = 'succeeded'::job_status)  <--------------- BOOOOO
                                                   Rows Removed by Filter: 16865
                                       ->  Sort  (cost=1349.52..1353.40 rows=1553 width=16) (actual time=9.377..17.009 rows=101995 loops=3)
                                             Sort Key: (((connection.id)::character varying(255))::text)
                                             Sort Method: quicksort  Memory: 270kB
                                             Worker 0:  Sort Method: quicksort  Memory: 270kB
                                             Worker 1:  Sort Method: quicksort  Memory: 270kB
                                             ->  Seq Scan on connection  (cost=0.00..1267.20 rows=1553 width=16) (actual time=0.055..4.929 rows=1572 loops=3)
                                                   Filter: (status = 'active'::status_type)  <--------------- BOOOOO
                                                   Rows Removed by Filter: 8899
 Planning Time: 0.440 ms
 Execution Time: 1144.759 ms
(42 rows)
```

## How
Add a new migration to create indices on the aforementioned columns.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_40_18_001__AddIndexToAttemptsAndJobsStatus.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package io.airbyte.db.instance.jobs.migrations;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
```
Comment (davinchia) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/jobs/migrations/V0_40_18_001__AddIndexToAttemptsAndJobsStatus.java:10]: get rid of comment

File: airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_18_003__AddIndexToConnectionStatus.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package io.airbyte.db.instance.configs.migrations;
+
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.impl.DSL;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+// TODO: update migration description in the class name
```
Comment (davinchia) [inline airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_18_003__AddIndexToConnectionStatus.java:10]: get rid of comment

**Review by davinchia — APPROVED** [review]
Great - the .class files should be gotten rid off before we merge in!

Can you also update the PR description so there are breadcrumbs for later devs? The tricky thing with indexes is remembering to remove them after the schema has changed and they are no longer in use. Something linking to the various query analysers is sufficient.

**Comment by davinchia** [comment]
@perangel you need to run `SUB_BUILD=PLATFORM ./gradlew format` to resolve some formatting issues.

**Review by xiaohansong — APPROVED** [review]

**Pull request merged**